hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -52,19 +53,31 @@ def __aiter__(self) -> 'PageIterator':
         return self
 
     async def __anext__(self):
+        import hailtop.httpx  # pylint: disable=import-outside-toplevel,cyclic-import
+
         if self._page is None:
             assert 'pageToken' not in self._request_params
-            self._page = await retry_transient_errors(
-                self._client.get, self._path, params=self._request_params, **self._request_kwargs
-            )
+            try:
+                self._page = await retry_transient_errors(
+                    self._client.get, self._path, params=self._request_params, **self._request_kwargs
+                )
+            except hailtop.httpx.ClientResponseError as e:
+                if e.status == 403 and 'does not have storage.objects.list access' in e.message:
+                    warnings.warn(f'ignoring error {e.status}: {e.message}')
+                    raise NotADirectoryError
```

Comment:
I'm a little confused by this change. 
Firstly (and for GCP generally), is 403 not is a perfectly valid error and users should be made aware? 
Second, and without context of how a `PageIterator` is used, why would getting a 403 imply that the object is not a directory? This feels like we're making this too fine-tuned for the terra data repository.